torchchat/generate.py

```diff
@@ -553,12 +552,10 @@ def decode_n_tokens(
                     **sampling_kwargs,
                 )
                 input_pos += 1
-                new_tokens.append(next_token.clone())
-                callback(new_tokens[-1], done_generating=_i == num_new_tokens - 2)
-                if need_probs or next_prob is None:
```

Comment:
This was backwards. It should be: `if not need probs or next_prob is None:`. Otherwise you are saying, if you need the probabilities you are getting None, and if you don't need them you are getting them.